tools/base/fpgainfo/board.c

```diff
@@ -0,0 +1,413 @@
+		case 'G':
+			if (NULL == tmp_optarg)
+				break;
+			if (!strcmp("0", tmp_optarg)) {
+				group_num = 0;
+			}
+			else if (!strcmp("1", tmp_optarg)) {
+				group_num = 1;
+			}
+			else if (!strcmp("all", tmp_optarg)) {
+				group_num = -1;
+			}
+			else {
+				fprintf(stderr, "Invalid argument '%s' of option group\n",
+					tmp_optarg);
+				return -1;
+			}
+			break;
+
+		case 'h':   /* help */
+			phy_help();
+			return -1;
+
+		case ':':   /* missing option argument */
+			fprintf(stderr, "Missing option argument\n");
+			phy_help();
+			return -1;
+
+		case '?':
+		default:    /* invalid option */
+			fprintf(stderr, "Invalid cmdline options\n");
+			phy_help();
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+fpga_result phy_filter(fpga_properties *filter, int argc, char *argv[])
+{
+	fpga_result res = FPGA_INVALID_PARAM;
+
+	if (0 == parse_phy_args(argc, argv)) {
+		res = fpgaPropertiesSetObjectType(*filter, FPGA_DEVICE);
+		fpgainfo_print_err("setting type to FPGA_DEVICE", res);
+	}
+	return res;
+}
+
+fpga_result phy_command(fpga_token *tokens, int num_tokens, int argc,
+	char *argv[])
+{
+	(void)argc;
+	(void)argv;
+	fpga_result res = FPGA_OK;
+	fpga_properties props;
+
+	int i = 0;
+	for (i = 0; i < num_tokens; ++i) {
+		res = fpgaGetProperties(tokens[i], &props);
+		ON_FPGAINFO_ERR_GOTO(res, out_destroy, "reading properties from token");
+
+		phy_group_info(tokens[i]);
+		fpgaDestroyProperties(&props);
+	}
+
+	return res;
+
+out_destroy:
+	fpgaDestroyProperties(&props);
+	return res;
+}
+
+
+// prints board version info
+fpga_result fpgainfo_board_info(fpga_token token)
+{
+	fpga_result res        = FPGA_OK;
+	void* dl_hanlde        = NULL;
+
+	// Board version
+	fpga_result(*print_board_info)(fpga_token token);
+
+	res = load_board_plugin(token, &dl_hanlde);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to load board plugin\n");
+		goto out;
+	}
+
+	print_board_info = dlsym(dl_hanlde, "print_board_info");
+	if (print_board_info) {
+		res = print_board_info(token);
+	}
+
+out:
+	return res;
+}
+
+// Prints mac info
+fpga_result mac_info(fpga_token token)
+{
+	fpga_result res       = FPGA_OK;
+	void* dl_hanlde       = NULL;
+
+	// mack
+	fpga_result(*print_mac_info)(fpga_token token);
+
+	res = load_board_plugin(token, &dl_hanlde);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to load board plugin\n");
+		goto out;
+	}
+
+	print_mac_info = dlsym(dl_hanlde, "print_mac_info");
+	if (print_mac_info) {
+		res = print_mac_info(token);
+	}
+
+out:
+	return res;
+}
+
+// prints PHY group info
+fpga_result phy_group_info(fpga_token token)
+{
+	fpga_result res         = FPGA_OK;
+	void* dl_hanlde         = NULL;
+
+	// phy group info
+	fpga_result(*print_phy_info)(fpga_token token);
+
+	res = load_board_plugin(token, &dl_hanlde);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to load board plugin\n");
+		goto out;
+	}
+
+	print_phy_info = dlsym(dl_hanlde, "print_phy_info");
+	if (print_phy_info) {
+		res = print_phy_info(token);

```

Comment:
Print an error / return an error code if the entry point isn't found.

Reply:
fixed